scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -723,6 +725,7 @@ private void processApplicationDataRecord(final Record record) {
 	private void handleApplicationMessage(ApplicationMessage message, DTLSSession session) {
 		if (messageHandler != null) {
 			DtlsCorrelationContext context = new DtlsCorrelationContext(
+					session.getPeerIdentity().getName(),
```

Comment:
There is no way to choose what to use as endpoint Identity.
What happens for unauthenticated peer ?